### PR TITLE
add Omnibus F4 SD page

### LINKED ISSUE #200: Document Omnibus F4 Pro Autopilot
As sub page of [Pixhawk Series](https://docs.px4.io/en/flight_controller/pixhawk_series.html).

Discussed: 
https://github.com/PX4/Firmware/pull/9310#issuecomment-381444412
https://github.com/PX4/Firmware/pull/9310#issuecomment-381475895

### PR DESCRIPTION
**Pull request by bkueng**
@nathantsoi:
- Can you provide the links for the Omnibus boards that you successfully tested?
- I added the omnibus picture from your blog, is that ok?

Fixes  #200

### CODE REVIEW

**Comment by nathantsoi** [comment]
Thanks for getting this rolling @bkueng! Great work!

I added pinout details and images here: https://github.com/PX4/px4_user_guide/pull/293

File: en/flight_controller/omnibus_f4_sd.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Omnibus F4 SD
+The Omnibus F4 SD is a board designed for racers. In contrast to a typical racer
+board it has some additional features, such as an SD card or a faster CPU.
```
Comment (mhkabir) [inline en/flight_controller/omnibus_f4_sd.md]: *and* a faster CPU.

File: en/flight_controller/omnibus_f4_sd.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+These are the main differences compared to a Pixracer:
+- Lower price
+- Less IO ports (though it's still possible to attach a GPS or a Flow sensor for example)
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: Fewer (fewer is used when discussing countable things, while less is used for singular mass nouns.)

File: en/flight_controller/omnibus_f4_sd.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+* Futaba S.BUS and S.BUS2 / Spektrum DSM2 and DSMX / Graupner SUMD / PPM input / Yuneec ST24
+* OneShot PWM out (configurable)
+* built-in current sensor
+* built-in OSD chip (AB7456 via SPI, not supported yet)
+
+
+## Where to Buy
+
+The board is produced by different vendors, with some variations (e.g. with or without a barometer).
+If the Betaflight OMNIBUSF4SD target can be used, then it should work with PX4 as well.
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: What is the "Betaflight OMNIBUSF4SD target". As a PX4 developer I don't know what this is, and how I would test it.

File: en/flight_controller/omnibus_f4_sd.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+The *Omnibus F4 SD* is a controller board designed for racers. 
+In contrast to a typical racer board it has some additional features, such as an SD card and a faster CPU.
+
+These are the main differences compared to a [Pixracer](../flight_controller/pixracer.md):
+- Lower price
+- Fewer IO ports (though it's still possible to attach a GPS or a Flow sensor for example)
+- Less RAM (192 KB vs. 256 KB) and FLASH (1 MB vs. 2 MB)
+- Same board dimensions as a *Pixracer*, but slightly smaller form factor (because it has less connectors)
+
+> **Tip** All the usual PX4 features can still be used for your racer!
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: FYI slightly modified this to make it clear that you can still use PX4 features. We could instead state what these limitations mean - fewer concurrently connected peripherals, there isn't much room for additional software in flash or RAM. I figure that the racer audience doesn't care about this.

**Comment by hamishwillee** [comment]
@bkueng I have subedited and added some notes. This is good as far as it goes and could be merged.

IMO this would vastly be improved by having:
1. A connector diagram like this: https://docs.px4.io/en/flight_controller/pixhawk.html#connectors
1.  Expanded pinout doc information like this: https://docs.px4.io/en/flight_controller/pixhawk.html#pinouts

In addition, I am a huge fan of wiring quickstarts - [like this](https://docs.px4.io/en/assembly/quick_start_pixhawk4.html). They make adoption much easier because they explain the minimal wiring configuration and also the recommended wiring configuration.

**Comment by nathantsoi** [comment]
i added a bunch of stuff @hamishwillee, maybe we can merge that into this branch via https://github.com/PX4/px4_user_guide/pull/293 and see what else needs to be added?

**Comment by hamishwillee** [comment]
@nathantsoi Yes, that stuff in #293 all looks good. Can you do a first pass merge? I think that addresses my first 2 points https://github.com/PX4/px4_user_guide/pull/289#issuecomment-402569518 - but would still be excellent to have a quickstart.

**Comment by hamishwillee** [comment]
Alternatively I can merge this, and you can then rebase against master?

File: en/flight_controller/omnibus_f4_sd.md (hunk 3 above)
Comment (bkueng) [reply en/flight_controller/omnibus_f4_sd.md]: Basically it just means that if you search for 'OMNIBUSF4SD' on the product page and find it, you're good to go.

Comment (hamishwillee) [reply en/flight_controller/omnibus_f4_sd.md]: It should probably say that then :-)

**Comment by bkueng** [comment]
@hamishwillee up to the 'Betaflight OMNIBUSF4SD target' this is good to go.

File: en/flight_controller/omnibus_f4_sd.md (hunk 3 above)
Comment (bkueng) [reply en/flight_controller/omnibus_f4_sd.md]: Right, do you want to do the rewrite?

Comment (hamishwillee) [reply en/flight_controller/omnibus_f4_sd.md]: "want" would be overstating it :-). Done in https://github.com/PX4/px4_user_guide/pull/289/commits/532c0042732698521e21037492fe536e19483efb

**Comment by hamishwillee** [comment]
@bkueng @nathantsoi  FYI, 
- I moved the silkscreens under own heading "Connectors" (that is why they are interesting) rather than "where to buy". 
- Changed image display to markdown format rather than HTML. This is a matter of taste, but generally I only use HTML if I have to for images - ie if I want to scale them.
- Moved some tips around to better organise info.

File: en/flight_controller/omnibus_f4_sd.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+### UARTs
+
+- UART6: GPS port
+  - TX: MCU pin PC6
+  - RX: MCU pin PC7
+
+  - Airbot Omnibus F4 SD Pinout is on Port J10 (TX6/RX6):
+
+  ![Omnibus F4 SD UART6](../../assets/flight_controller/omnibus_f4_sd/uart6.jpg)
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: Is the purpose of this image to show the relevant pins? If so, perhaps circle them in red? Once that done I would crop this image to minimum useful size.

File: en/flight_controller/omnibus_f4_sd.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  - TX: MCU pin PA0
+  - RX: MCU pin PA1
+  - 57600 baud
+  - It can be configured via [SYS_COMPANION](../advanced_config/parameter_reference.md#SYS_COMPANION) parameter.
+
+  - Airbot Omnibus F4 SD Pinout:
+    - TX: RSSI pin
+    - RX: PWM out 5
+
+  ![Omnibus F4 SD UART4](../../assets/flight_controller/omnibus_f4_sd/uart4.jpg)
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: I don't understand what I am supposed to be seeing in these images - perhaps label the bits of interest?

File: en/flight_controller/omnibus_f4_sd.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+There is one I2C port available via:
+- TX: MCU pin PB10
+- RX: MCU pin PB11
+
+> **Note** You will need external pullups on both signals (clock and data).
+> You can use 2.2k pullups for example to attach an external mag.
+
+  - Airbot Omnibus F4 SD Pinout is on Port J10 (SCL [clock] / SCA [data]):
+<img src="../../assets/flight_controller/omnibus_f4_sd/uart6.jpg" title="Omnibus F4 SD UART6" />
```
Comment (hamishwillee) [inline en/flight_controller/omnibus_f4_sd.md]: - If this image belongs to previous bullet point, it should be indented to align with the  text.
- Again, what is this image supposed to show. I would highlight the port of interest - e.g. red box around it.

**Comment by hamishwillee** [comment]
@bkueng OK, few more comments inline about images. I am done. 

This is much better/usable/useful. IMO it would still be good (but not essential) if 
- we had consistent pinout docs *format* (e.g. like [pixhawk>pinouts](https://docs.px4.io/en/flight_controller/pixhawk.html#pinouts)
- We could have a quickstart - [like this](https://docs.px4.io/en/assembly/quick_start_pixhawk4.html) to ease adoption.

File: en/flight_controller/omnibus_f4_sd.md (hunk 3 above)
Comment (bkueng) [reply en/flight_controller/omnibus_f4_sd.md]: Haha, thanks!

**Comment by bkueng** [comment]
> I moved the silkscreens under own heading "Connectors" (that is why they are interesting) rather than "where to buy".

I think the reason why it was there is that it specifically applies to the Airbot Omnibus F4 SD.

> we had consistent pinout docs format

This is more difficult here since we don't necessarily have connectors on the board, but only the pins that you solder directly individual cables on.

@nathantsoi is best to answer the other questions.

**Comment by hamishwillee** [comment]
> I think the reason why it was there is that it specifically applies to the Airbot Omnibus F4 SD.

A diagram showing the connectors is a useful thing. Are you saying that variants will differ enough that the silkscreen can't serve that purpose for all boards?

**Comment by bkueng** [comment]
> A diagram showing the connectors is a useful thing. Are you saying that variants will differ enough that the silkscreen can't serve that purpose for all boards?

Yes, the hobbywing board already looks quite different for example.

**Comment by hamishwillee** [comment]
@bkueng Thanks. 

>> A diagram showing the connectors is a useful thing. Are you saying that variants will differ enough that the silkscreen can't serve that purpose for all boards?

> Yes, the hobbywing board already looks quite different for example.

Thanks. It still makes sense that this not be in the "Where to buy" section. I have just added another heading level to make these images board specific: https://github.com/PX4/px4_user_guide/pull/289/commits/0eaeacd984df8d3934034f041d6a9f94715af00b

Can we get a silkscreen for the hobbyking version there too?

I'm OK with this doc to be merged except for the previous inline comments about images not being clear about "what they apply to". A little bit of red around the connectors of interest and maybe a label would help.

**Comment by bkueng** [comment]
> Can we get a silkscreen for the hobbyking version there too?

Done. Source: http://www.hobbywing.com/products/enpdf/XRotorFlightController.pdf

**Comment by hamishwillee** [comment]
Nice one. Should we merge and then hope @nathantsoi Can provide better markup on the connection diagrams later?

**Comment by bkueng** [comment]
Yep, let's get it in. Thanks for the help @hamishwillee & @nathantsoi

**Pull request merged**